packages/@aws-cdk/aws-kinesis/test/stream.test.ts

```diff
@@ -431,6 +511,17 @@ describe('Kinesis data streams', () => {
     });
   });
 
+  test('throws when using shardCount with on-demand streamMode', () => {
+    const stack = new Stack();
+
+    expect(() => {
+      new Stream(stack, 'MyStream', {
+        shardCount: 2,
+        streamMode: StreamMode.ON_DEMAND,
+      });
+    }).toThrow('streamMode must be set to PROVISIONED (default) when specifying shardCount');
```

Comment:
You can use `streamMode must be set to ${StreamMode.PROVISIONED} (default) when specifying shardCount` to keep the value of the enum even if it is changed in the future.